### PR TITLE
Encodings and bug fixes

### PR DESCRIPTION
**Pull request by MW-88**

### CODE REVIEW

File: src/selected_words_counter/extract_files.py [hunk 1]
```diff
@@ -58,11 +61,10 @@ def process_and_save_file(
         )
         if verbose:
             print(a_output_name)
-        with open(a_output_name, "w") as file:
+        with open(a_output_name, "w", encoding="utf-8") as file:
             file.write(str(text_content))
-
     except Exception as e:
-        print(f"Error processing {afilepath}: {e}")
+        print("Error with writing file: " + e)
```
Comment (Pieter-Kouyzer) [inline src/selected_words_counter/extract_files.py:67]: Probably still worthwhile to print the filepath as well.

File: src/selected_words_counter/extract_files.py [hunk 2]
```diff
@@ -86,19 +88,26 @@ def extract_msg_attachments(atarget_dir):
...
-                    att = msg.attachments[item]
-                    msg.attachments[item].save(
-                        customPath=a_output_directory, customFilename=att.longFilename
-                    )
-        except Exception as e:
-            print(e)
+                if len(msg.attachments) > 0:
+                    a_output_directory = make_dir_from_filename(afilepath)
+
+                    for item in range(0, len(msg.attachments)):
```
Comment (Pieter-Kouyzer) [inline src/selected_words_counter/extract_files.py:100]: Shouldn't you check the encodings for the attachments as well?

File: src/selected_words_counter/extract_files.py [hunk 3]
```diff
@@ -86,19 +88,26 @@ def extract_msg_attachments(atarget_dir):
...
+                        att = msg.attachments[item]
+                        msg.attachments[item].save(
+                            customPath=a_output_directory,
+                            customFilename=att.longFilename,
+                        )
+                return 0
+            except Exception as e:
+                continue
+        print(f"Failed to read {afilepath} with all tested encodings.")
```
Comment (Pieter-Kouyzer) [inline src/selected_words_counter/extract_files.py:109]: Maybe check in the main code where this function is called if it succeeeded? By the way returning 0 is usually not a sign that it was successfull. Maybe work with True and False instead? True meaning successful, False not successful.

File: src/selected_words_counter/extract_files.py [hunk 4]
```diff
@@ -196,18 +203,22 @@ def run(atarget_dir, atarget_dir_extracted, amulti_thread=False):
...
-                page_text = page.extract_text()
-
-                # Handle encoding issues with utf-8-sig and replace errors
-                if page_text:
-                    text += page_text.encode("utf-8-sig", errors="replace").decode(
-                        "utf-8-sig", errors="replace"
-                    )
+                text += (
+                    page.extract_text()
+                    .encode("utf-8", errors="ignore")
```
Comment (Pieter-Kouyzer) [inline src/selected_words_counter/extract_files.py:219]: We shouldn't check for the other encodings?

File: src/selected_words_counter/extract_files.py [hunk 5]
```diff
@@ -293,7 +312,7 @@ def process_file(a_file_path):
             elif "pptx" in file_type:
                 a_content = read_pptx(a_file_path)
         except Exception as e:
-            print("Errow with " + a_file_path)
+            print("Errow with Reading " + a_file_path)
```
Comment (Pieter-Kouyzer) [inline src/selected_words_counter/extract_files.py:315]: *Error

File: tests/test_selected_word_counter.py [hunk 6]
```diff
@@ -81,6 +82,11 @@ def test_extract_files_run():
     finally:
         shutil.rmtree(test_dir_converted)
 
+        for item in os.listdir(test_dir):
```
Comment (Pieter-Kouyzer) [inline tests/test_selected_word_counter.py:85]: Why is this necessary? Is this not deleting the test_files?

File: tests/test_selected_word_counter.py [hunk 7]
```diff
@@ -181,3 +195,80 @@ def test_no_extract():
                 ), f"{filepath} should contain {word} {count} times"
     finally:
         os.remove(result_output + ".xlsx")
+        for item in os.listdir(test_dir):
+            item_path = os.path.join(test_dir, item)
+            if os.path.isdir(item_path):
+                shutil.rmtree(item_path)
+
+
+def test_count_files_nl():
```
Comment (Pieter-Kouyzer) [inline tests/test_selected_word_counter.py:204]: test_count_files and test_count_files_nl seem very alike in code, although different in content. Maybe it's easy to remove some code duplication?

File: make_requirements.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import pkg_resources
+
+# List installed packages in the current environment
+installed_packages = pkg_resources.working_set
```
[the remark this answers is not in the document]
Comment (MW-88) [reply make_requirements.py]: removed this file

File: src/selected_words_counter/counting.py [hunk 9]
```diff
@@ -45,6 +47,12 @@ def word_counting_in_files(aword_list, afilepaths, exact_match=False):
                         sum(1 for _ in re.finditer(r"\b%s\b" % re.escape(word), text))
                         for word in aword_list
                     ]
+                elif white_spaces:
```
[the remark this answers is not in the document]
Comment (MW-88) [reply src/selected_words_counter/counting.py:50]: encoding issue's some returning words like 'c a t' instead of cat.
This should now be fixed but i thought i would let it be a optional option

File: src/selected_words_counter/extract_files.py (hunk 2 above)
Comment (MW-88) [reply src/selected_words_counter/extract_files.py:100]: Don't think so because reading in the whole email should also encode? Not sure though worth to check out

Comment (MW-88) [reply src/selected_words_counter/extract_files.py:103]: for now in a other pr

File: config.py [hunk 10]
```diff
@@ -15,7 +24,7 @@
 # Wether to delete converted data.
 keep_extracted = False
 # Wether to multithread read in of files, a possible speed up but currently not always working
-multi_thread = False
+multi_thread = True
```
[the remark this answers is not in the document]
Comment (MW-88) [reply config.py:27]: Fixed

File: src/selected_words_counter/extract_files.py (hunk 3 above)
Comment (MW-88) [reply src/selected_words_counter/extract_files.py:112]: returning false now, it is a bit ugly like this indeed but it works will fix in a furter PR

File: tests/test_selected_word_counter.py (hunk 6 above)
Comment (MW-88) [reply tests/test_selected_word_counter.py:91]: only the test folder of which the original files were extracted

Comment (MW-88) [reply tests/test_selected_word_counter.py:91]: tested this multiple times so i am sure that it works

File: config.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (Pieter-Kouyzer) [reply config.py:27]: I still see this as True?

File: tests/test_pdf.ipynb
Comment (Pieter-Kouyzer) [inline tests/test_pdf.ipynb:1]: Please more this one before you finish up!

**Review by Pieter-Kouyzer — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 11]
```diff
@@ -1,12 +1,16 @@
+chardet==3.0.4
 extract_msg==0.28.7
+huggingface_hub==0.26.2
```
[the remark this answers is not in the document]
Comment (MW-88) [reply requirements.txt:7]: fixed old requirements
